### PR TITLE
Fix #3747 - FQDN model in ometa api

### LINKED ISSUE #3747: Fix py-tests CI
**Affected module**
python ingestion

**Describe the bug**
```
=========================== short test summary info ============================
FAILED ingestion/tests/integration/ometa/test_ometa_chart_api.py::OMetaChartTest::test_delete
FAILED ingestion/tests/integration/ometa/test_ometa_chart_api.py::OMetaChartTest::test_get_entity_ref
FAILED ingestion/tests/integration/ometa/test_ometa_chart_api.py::OMetaChartTest::test_get_entity_version
FAILED ingestion/tests/integration/ometa/test_ometa_chart_api.py::OMetaChartTest::test_get_id
FAILED ingestion/tests/integration/ometa/test_ometa_chart_api.py::OMetaChartTest::test_get_name
FAILED ingestion/tests/integration/ometa/test_ometa_chart_api.py::OMetaChartTest::test_list_versions
FAILED ingestion/tests/integration/ometa/test_ometa_dashboard_api.py::OMetaDashboardTest::test_delete
FAILED ingestion/tests/integration/ometa/test_ometa_dashboard_api.py::OMetaDashboardTest::test_get_entity_ref
FAILED ingestion/tests/integration/ometa/test_ometa_dashboard_api.py::OMetaDashboardTest::test_get_entity_version
FAILED ingestion/tests/integration/ometa/test_ometa_dashboard_api.py::OMetaDashboardTest::test_get_id
FAILED ingestion/tests/integration/ometa/test_ometa_dashboard_api.py::OMetaDashboardTest::test_get_name
FAILED ingestion/tests/integration/ometa/test_ometa_dashboard_api.py::OMetaDashboardTest::test_list_versions
FAILED ingestion/tests/integration/ometa/test_ometa_database_service_api.py::OMetaDatabaseServiceTest::test_delete
FAILED ingestion/tests/integration/ometa/test_ometa_database_service_api.py::OMetaDatabaseServiceTest::test_get_id
FAILED ingestion/tests/integration/ometa/test_ometa_database_service_api.py::OMetaDatabaseServiceTest::test_get_name
FAILED ingestion/tests/integration/ometa/test_ometa_database_service_api.py::OMetaDatabaseServiceTest::test_list_versions
FAILED ingestion/tests/integration/ometa/test_ometa_model_api.py::OMetaModelTest::test_delete
FAILED ingestion/tests/integration/ometa/test_ometa_model_api.py::OMetaModelTest::test_get_entity_ref
FAILED ingestion/tests/integration/ometa/test_ometa_model_api.py::OMetaModelTest::test_get_entity_version
FAILED ingestion/tests/integration/ometa/test_ometa_model_api.py::OMetaModelTest::test_get_id
FAILED ingestion/tests/integration/ometa/test_ometa_model_api.py::OMetaModelTest::test_get_name
FAILED ingestion/tests/integration/ometa/test_ometa_model_api.py::OMetaModelTest::test_list_versions
FAILED ingestion/tests/integration/ometa/test_ometa_model_api.py::OMetaModelTest::test_mlmodel_properties
FAILED ingestion/tests/integration/ometa/test_ometa_model_api.py::OMetaModelTest::test_update
================== 24 failed, 70 passed, 2 warnings in 6.80s ===================
```

**To Reproduce**

Screenshots or steps to reproduce

**Expected behavior**

**Version:**
 - Python version:
 - OpenMetadata version: [e.g. 0.8]
 - OpenMetadata Ingestion package version: [e.g. `openmetadata-ingestion[docker]==XYZ`]

**Additional context**

### PR DESCRIPTION
**Pull request by pmbrull**
### Describe your changes :

This PR fixes https://github.com/open-metadata/OpenMetadata/issues/3747

#
### Type of change :

- [x] Bug fix
- [x] Improvement
- [x] New feature
- [] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [] Documentation

#
### Frontend Preview (Screenshots) :
<p align="center">For frontend related change, please link screenshots of your changes preview! Optional for backend related changes.
</p>

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/open-source-community/developer) document.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#

### CODE REVIEW

File: ingestion/src/metadata/ingestion/ometa/utils.py [hunk 1]
```diff
@@ -66,3 +67,17 @@ def uuid_to_str(entity_id: Union[str, basic.Uuid]) -> str:
         return str(entity_id.__root__)
 
     return entity_id
+
+
+def fqdn_to_str(fqdn: Union[str, FullyQualifiedEntityName]) -> str:
```
Comment (pmbrull) [inline ingestion/src/metadata/ingestion/ometa/utils.py:72]: Here I followed the same approach as with UUID. However, if after EntityName PR we have a similar scenario, we can just prepare a generalized method to handle either str or models with __root__.

@ayush-shah feel free to give this a more elegant go

Comment (ayush-shah) [reply ingestion/src/metadata/ingestion/ometa/utils.py:72]: I think we can merge 2 methods into one - i.e. UUID method uuid_to_str and fqdn_to_str can be merged using a Union for both models or 
list of models that need __root__

Comment (ayush-shah) [reply ingestion/src/metadata/ingestion/ometa/utils.py:72]: cc: @pmbrull

Comment (pmbrull) [reply ingestion/src/metadata/ingestion/ometa/utils.py]: ok got a stab at it as I also needed to fix the EntityName Thanks

File: ingestion/src/metadata/ingestion/ometa/utils.py [hunk 2]
```diff
@@ -54,15 +55,19 @@ def get_entity_type(
...
-    if isinstance(entity_id, basic.Uuid):
-        return str(entity_id.__root__)
+    return str(arg)
 
-    return entity_id
+
+@model_str.register(basic.Uuid)
+@model_str.register(basic.FullyQualifiedEntityName)
+@model_str.register(basic.EntityName)
+def _(arg) -> str:
```
Comment (pmbrull) [inline ingestion/src/metadata/ingestion/ometa/utils.py:69]: This lets us have a bit more control on the methods we need to extract __root__ from, as IDK if I'd rather have an if/else checking for an attribute which can hide some issues. Let me know what you think

**Review by ayush-shah — APPROVED** [review]

**Pull request merged**